src/libs/sfdk/dockervirtualmachine_p.h

```diff
@@ -0,0 +1,109 @@
+    static void fetchRegisteredVirtualMachines(const QObject *context,
+            const Functor<const QStringList &, bool> &functor);
+    static bool isAvailable();
+
+private:
+    Q_DISABLE_COPY(DockerVirtualMachine)
+    Q_DECLARE_PRIVATE(DockerVirtualMachine)
+};
+
+class DockerVirtualMachinePrivate : public VirtualMachinePrivate
+{
+    Q_DECLARE_PUBLIC(DockerVirtualMachine)
+
+public:
+    using VirtualMachinePrivate::VirtualMachinePrivate;
+
+    void fetchInfo(VirtualMachineInfo::ExtraInfos extraInfo, const QObject *context,
+            const Functor<const VirtualMachineInfo &, bool> &functor) const override;
+
+    void start(const QObject *context, const Functor<bool> &functor) override;
+    void stop(const QObject *context, const Functor<bool> &functor) override;
+    void probe(const QObject *context,
+            const Functor<BasicState, bool> &functor) const override;
+
+    void setVideoMode(const QSize &size, int depth, const QString &deviceModelName,
+            Qt::Orientation orientation, int scaleDownFactor, const QObject *context,
+            const Functor<bool> &functor) override;
+
+protected:
+    void doSetMemorySizeMb(int memorySizeMb, const QObject *context,
+            const Functor<bool> &functor) override;
+    void doSetCpuCount(int cpuCount, const QObject *context,
+            const Functor<bool> &functor) override;
+    void doSetStorageSizeMb(int storageSizeMb, const QObject *context,
+            const Functor<bool> &functor) override;
+    void doSetSharedPath(SharedPath which, const Utils::FileName &path, const QObject *context,
+            const Functor<bool> &functor) override;
+
+    void doAddPortForwarding(const QString &ruleName,
+            const QString &protocol, quint16 hostPort, quint16 emulatorVmPort,
+            const QObject *context, const Functor<bool> &functor) override;
+    void doRemovePortForwarding(const QString &ruleName,
+            const QObject *context, const Functor<bool> &functor) override;
+    void doSetReservedPortForwarding(ReservedPort which, quint16 port,
+            const QObject *context, const Functor<bool> &functor) override;
+    void doSetReservedPortListForwarding(ReservedPortList which, const QList<Utils::Port> &ports,
+            const QObject *context, const Functor<const QMap<QString, quint16> &, bool> &functor)
+            override;
+
+    void doRestoreSnapshot(const QString &snapshotName, const QObject *context,
+            const Functor<bool> &functor) override;
+
+private:
+    VirtualMachineInfo virtualMachineInfoFromOutput(const QByteArray& output) const;
+
+    static QString labelFor(SharedPath which);
+    static QString labelFor(ReservedPort which);
+
+    static QString dockerPath();
+
+    static QStringList listedImages(const QString &output);
+
+    void buildWithLabel(const QString& key, const QString& value,
+                        const QObject *context, const Functor<bool> &functor);
+
+    mutable QString m_containerId;
```

Comment:
It is now in the private class, so `m_` should not be used.

Reply:
fixed